tests/drivers/opamp/opamp_api/boards/frdm_mcxa156.overlay

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	zephyr,user {
+		opamp = <&opamp0>;
+	};
+};
+
+&opamp0 {
```

Comment:
If you consider filtering approach from exti PR https://github.com/zephyrproject-rtos/zephyr/pull/85508 you'll be able to add only one general `test_opamp.dtsi` file,  which will hold for most of the devices.